File: core/x/javaapi/api/src/main/java/viaduct/java/api/types/Edge.java
package viaduct.java.api.types;

/**
 * A wrapper around a node that also carries a {@code cursor} encoding the item's position in the
 * list. Clients pass the cursor back as {@code after}/{@code before} to resume pagination from that
 * point.
 *
 * <p>{@link viaduct.java.api.internal.ConnectionBuilder#fromSlice} and {@link
 * viaduct.java.api.internal.ConnectionBuilder#fromList} set {@code node} and {@code cursor}
 * automatically. Use {@link viaduct.java.api.internal.ConnectionBuilder#fromEdges} when your edge
 * type has additional custom fields.
 *
 * <p>Java equivalent of Kotlin's {@code viaduct.api.types.Edge}.
 *
 * @param <N> the type of node this edge contains
 * @see Connection
 */
public interface Edge<N> extends GraphQLObject {}

File: core/tenant/api/src/main/kotlin/viaduct/api/types/Edge.kt
package viaduct.api.types

import viaduct.apiannotations.ExperimentalApi

/**
 * A wrapper around a node that also carries a `cursor` encoding the item's position in the list.
 * Clients pass the cursor back as `after`/`before` to resume pagination from that point.
 *
 * [viaduct.api.internal.ConnectionBuilder.fromSlice] and [viaduct.api.internal.ConnectionBuilder.fromList]
 * set `node` and `cursor` automatically. Use [viaduct.api.internal.ConnectionBuilder.fromEdges]
 * when your edge type has additional custom fields.
 *
 * @param N The type of node this edge contains.
 * @see Connection
 */
@ExperimentalApi
interface Edge<N> : Object
